### ISSUE TITLE
SpanMultiTermQueryBuilderTests.testToQueryInnerTermQuery reproducibly fails in 7.x

### ISSUE THREAD

**Issue by imotov** [elastic/elasticsearch#58199]
**Build scan**:
https://elasticsearch-ci.elastic.co/job/elastic+elasticsearch+pull-request-1/1560/

**Repro line**:
```
./gradlew ':server:test' --tests "org.elasticsearch.index.query.SpanMultiTermQueryBuilderTests.testToQueryInnerTermQuery" -Dtests.seed=92425189D86F7872 -Dtests.security.manager=true -Dtests.locale=uk -Dtests.timezone=America/St_Thomas -Druntime.java=8
```

**Reproduces locally?**:
yes

**Applicable branches**:
7.x

**Failure history**:
Couldn't find any other recent failures

**Failure excerpt**:
```
ite: Test class org.elasticsearch.index.query.SpanMultiTermQueryBuilderTests
  1> [2020-06-16T16:17:22,534][INFO ][o.e.i.q.SpanMultiTermQueryBuilderTests] [testToQueryInnerTermQuery] before test
  1> [2020-06-16T16:17:22,589][INFO ][o.e.p.PluginsService     ] [testToQueryInnerTermQuery] no modules loaded
  1> [2020-06-16T16:17:22,589][INFO ][o.e.p.PluginsService     ] [testToQueryInnerTermQuery] loaded plugin [org.elasticsearch.test.TestGeoShapeFieldMapperPlugin]
  1> [2020-06-16T16:17:23,605][INFO ][o.e.p.PluginsService     ] [testToQueryInnerTermQuery] no modules loaded
  1> [2020-06-16T16:17:23,606][INFO ][o.e.p.PluginsService     ] [testToQueryInnerTermQuery] loaded plugin [org.elasticsearch.test.TestGeoShapeFieldMapperPlugin]
  1> [2020-06-16T16:17:23,625][INFO ][o.e.i.q.SpanMultiTermQueryBuilderTests] [testToQueryInnerTermQuery] after test
  2> REPRODUCE WITH: ./gradlew ':server:test' --tests "org.elasticsearch.index.query.SpanMultiTermQueryBuilderTests.testToQueryInnerTermQuery" -Dtests.seed=92425189D86F7872 -Dtests.security.manager=true -Dtests.locale=uk -Dtests.timezone=America/St_Thomas -Druntime.java=8
  2> java.lang.AssertionError: 
    Expected: an instance of org.apache.lucene.search.spans.SpanMultiTermQueryWrapper
         but: <mask(prefix_field._index_prefix:foo) as prefix_field> is a org.apache.lucene.search.spans.FieldMaskingSpanQuery
        at __randomizedtesting.SeedInfo.seed([92425189D86F7872:344C43FEE62E883A]:0)
        at org.hamcrest.MatcherAssert.assertThat(MatcherAssert.java:18)
        at org.junit.Assert.assertThat(Assert.java:956)
        at org.junit.Assert.assertThat(Assert.java:923)
        at org.elasticsearch.index.query.SpanMultiTermQueryBuilderTests.testToQueryInnerTermQuery(SpanMultiTermQueryBuilderTests.java:206)
```

**Comment by romseygeek** [comment]
The test checks to see if the index has been created before version 6.4, in which case index prefixes are unavailable and so it expects to see a span multi-term wrapper.  However, the production code doesn't bother with checking for versions, because if the field in question is configured with `index_prefixes` then it knows that it must have been created post 6.4 (you can't merge in a new `index_prefixes` configuration).  I think we just need to adjust the test to not bother checking the version?

**Comment by mark-vieira** [comment]
This is continuing to fail so I've muted it in in `7.x` with ef8899b1307ba508b01e86a18ff6e4c1dacbede7

**Comment by romseygeek** [comment]
Closed by #62833
